server/portal/apps/workspace/api/views.py

```diff
@@ -239,24 +243,36 @@ def post(self, request, *args, **kwargs):
                         )
 
             if settings.DEBUG:
-                wh_base_url = settings.WH_BASE_URL + '/webhooks/'
+                wh_base_url = settings.WH_BASE_URL + reverse('webhooks:interactive_wh_handler')
                 jobs_wh_url = settings.WH_BASE_URL + reverse('webhooks:jobs_wh_handler')
             else:
-                wh_base_url = request.build_absolute_uri('/webhooks/')
+                wh_base_url = request.build_absolute_uri(reverse('webhooks:interactive_wh_handler'))
                 jobs_wh_url = request.build_absolute_uri(reverse('webhooks:jobs_wh_handler'))
 
-            job_post['parameterSet']['envVariables'] = job_post['parameterSet'].get('envVariables', []) + [{'key': '_webhook_base_url', 'value':  wh_base_url}]
+            # Add additional data for interactive apps
+            if body.get('isInteractive'):
+                # Add webhook URL environment variable for interactive apps
+                job_post['parameterSet']['envVariables'] = job_post['parameterSet'].get('envVariables', []) + \
+                                                           [{'key': '_INTERACTIVE_WEBHOOK_URL', 'value':  wh_base_url}]
+
+                # Make sure $HOME/.tap directory exists for user when running interactive apps
+                execSystemId = job_post['execSystemId']
+                system = settings.PORTAL_EXEC_SYSTEMS.get(execSystemId)
+                tasdir = get_user_data(username)['homeDirectory']
+                if system:
+                    tapis.files.mkdir(systemId=execSystemId, path=f"{system['home_dir']}/{tasdir}/.tap")
```

Comment:
```suggestion
                    tapis.files.mkdir(systemId=execSystemId, path=f"{system['home_dir'].format(tasdir)}/.tap")

```

Reply:
I can submit the matlab job now. It completes successfully after a few minutes and didn't see the connection message 🤔